src/posix/net.cpp

```diff
@@ -1776,24 +1843,25 @@ void CurlHttpIO::post(HttpReq* req, const char* data, unsigned len)
             send_request(httpctx);
             return;
         }
+    }
+
+    if (dnsEntry && dnsEntry->ipv4.size() && !dnsEntry->isIPv4Expired())
+    {
+        LOG_debug << "DNS cache hit for " << httpctx->hostname << " (IPv4) " << dnsEntry->ipv4;
+        httpctx->isIPv6 = false;
+        httpctx->isCachedIp = true;
+        httpctx->hostip = dnsEntry->ipv4;
+        httpctx->ares_pending = 0;
+        send_request(httpctx);
+        return;
+    }
 
+    if (ipv6requestsenabled)
+    {
         httpctx->ares_pending++;
         LOG_debug << "Resolving IPv6 address for " << httpctx->hostname;
         ares_gethostbyname(ares, httpctx->hostname.c_str(), PF_INET6, ares_completed_callback, httpctx);
     }
-    else
-    {
-        if (dnsEntry && dnsEntry->ipv4.size() && !dnsEntry->isIPv4Expired())
-        {
-            LOG_debug << "DNS cache hit for " << httpctx->hostname << " (IPv4) " << dnsEntry->ipv4;
-            httpctx->isIPv6 = false;
-            httpctx->isCachedIp = true;
-            httpctx->hostip = dnsEntry->ipv4;
-            httpctx->ares_pending = 0;
-            send_request(httpctx);
-            return;
-        }
-    }
 
     LOG_debug << "Resolving IPv4 address for " << httpctx->hostname;
     ares_gethostbyname(ares, httpctx->hostname.c_str(), PF_INET, ares_completed_callback, httpctx);
```

Comment:
this code rearrangement prevents resending ipv4 DNS requests on every http request when ipv6 requests are enabled, but ipv6 lookups are not succeeding